### DISCUSSION TITLE
Request: how do I get SSR working in Astro?

### DISCUSSION THREAD

**Discussion by EdwardDrapkin** [tamagui/tamagui#2795] [General]
I'm trying to integrate Tamagui into Astro.

I've got everything working except SSR rendering - I get a warning about the client side generated to-hydrate HTML not matching the HTML it's replacing because of classnames.  It appears the TamaguiContext is not setting theme values when rendered server-side.  I know you guys have this all working in Next, and I can help do the legwork, but I have no idea where even to start solving this problem.

Thanks in advance!

**Comment by TotalLag** [comment]
@EdwardDrapkin can you share how you got this working? Trying to do similar things but SSG
